spec/ace.md

```diff
@@ -130,8 +130,69 @@ An executor MAY implement a "strict mode" where an image cannot run unless all i
+ 4. Values starting with `@` identify special wildcards and are scoped by the `/` separator. The `@appc/` scope is reserved for usage in this specification, implementations MAY provide additional wildcards in their own namespace (eg. `@implementation/wildcard`). The following special values are currently defined:
+  * `@appc/empty` represents the empty set. In the context of `seccomp-remove-set`, this wildcard masks all other values specified in `set` and can be effectively used to opt-in implementation-specific seccomp filtering.
+
+
+**Example:**
+```json
+"name": "os/linux/seccomp-remove-set",
+"value": {
+  "errno": "ENOTSUP",
+  "set": [
+    "chown",
+    "chmod"
+  ]
+}
+```
+
+In the example above, the process will not be allowed to invoke `chown(2)` and `chmod(2)` (and any other syscall in the implementation-specific blacklist). When invoked, such syscalls will immediately return a `ENOTSUP` error code. All other syscalls will behave as usual.
+
+#### os/linux/seccomp-retain-set
+
+* Scope: app
+
+**Parameters:**
+
+* **set** case-sensitive list of syscall names that will be whitelisted (ie. not blocked). All syscalls specified in this set MUST NOT be blocked. This set MAY be augmented by an implementation-specific default whitelist set (see notes below). Syscalls not specified in the union of these two sets MUST be blocked.
+* **errno** all-uppercase name of a single [errno code](http://man7.org/linux/man-pages/man3/errno.3.html) that will be returned by blocked system calls, instead of terminating. If missing or empty string, by default blocked syscalls MUST result in app termination via `SIGSYS` signal.
+
+**Notes:**
+ 1. The `os/linux/seccomp-retain-set` isolator cannot be used in conjunction with the `os/linux/seccomp-remove-set` isolator.
+ 2. When an app does not have any seccomp isolator (neither `os/linux/seccomp-remove-set` nor `os/linux/seccomp-retain-set` are specified), implementations MAY apply their own set of blocked syscalls.
+ 3. The implementation-specific default whitelist MUST be the minimum set of syscalls required for app life-cycle. For example, implementations may require the `exit(2)` syscall to be always allowed for clean app termination. This set MAY be empty.
+ 4. Values starting with `@` identify special wildcards and are scoped by the `/` separator. The `@appc/` top-level scope is reserved for usage in this specification. Implementations MAY provide additional wildcards in their own namespace (eg. `@implementation/wildcard`). The following special values are currently defined:
+  * `@appc/all` represents the set of all available syscalls. In the context of `seccomp-retain-set`, this wildcard masks all other values specified in `set` and can be effectively used to opt-out seccomp filtering.
```

Comment:
Two remarks here:
 * I think there should be in the spec some way for users to opt-out. We may reword this as a SHOULD and mention sensible security defaults or side-effects with other isolators, but keep it here.
 * From OOB discussion you are referring to [this systemd snippet](https://github.com/systemd/systemd/blob/master/src/nspawn/nspawn-seccomp.c#L39). It was introduced by [this commit](https://github.com/systemd/systemd/commit/9a71b1122c6e49dd9227f82b2f53837c7ea13019) and I think this is only partially relevant, as it further enforces normal capability behavior via seccomp (ie. directly blocking syscalls with EPERM when missing capabilities). No process termination there, and no visible effects if the app lacks the capability in the first place. I'd consider this an implementation detail of capabilities enforcing, which just happens to use seccomp but shouldn't practically hinder further filters setup.